### ISSUE TITLE
[Feature Request] Support multiple Notion Databases

### ISSUE THREAD

**Issue by d-walsh** [dvanoni/notero#119]
**Is your feature request related to a problem? Please describe.**
I have different Zotero Collections (on different topics) and I want them to be synced to different Notion Databases (I have different information in each database). Currently Notero only allow you to sync multiple collections to a single database.

**Describe the solution you'd like**
For each Notion Database I would like to select specific Zotero Collections that should be synced.

**Describe alternatives you've considered**
Installing different instances of Zotero with Notero configured differently on each..

**Additional context**

**Comment by dvanoni** [comment]
Hi @d-walsh, thanks for the feedback!

This was discussed previously in https://github.com/dvanoni/notero/issues/6#issuecomment-1012813567. The decision was made at the time to sync to a single database instead of multiple for a few reasons:

- Simplify the syncing implementation and preferences UI
- Avoid issues with managing multiple Notion links on a single Zotero item
- Have a single source of truth in Notion that more closely mirrors how Zotero stores data

To achieve an equivalent to syncing different collections to different databases, we proposed #30. The idea with that solution is that you'd still have a single Notion database, but you could create views of that database filtered to whichever collection(s) you want.

Would #30 and filtered database views provide an adequate solution for you?

**Comment by d-walsh** [comment]
Hi @dvanoni thanks for explaining that!

> Would https://github.com/dvanoni/notero/issues/30 and filtered database views provide an adequate solution for you?

Ideally I would have each collection Synced to a different database - because each collection for me is a completely different subject and I heavily customize each Notion Database. However it's not urgent and in the interm I think the solution proposed would be an improvement.

**Comment by zmek** [comment]
@dvavoni thanks for the brilliant work on Notero. I would also like to have the feature requested by @d-walsh - ie have multiple Notion databases each synced to different Zotero collections. However as you say the proposed interim solution should provide the same functionality

**Comment by KB22222** [comment]
Thank you for creating this! I agree with those who commented above, an option to create a separate database would be ideal. However, if that's not possible, the ability to input the Zotero folder name into notion would be great, so that I could at least sort by project in Notion.

**Comment by dvanoni** [comment]
@KB22222, you should be able to achieve this if you add a **multi-select** property named `Collections` to your Notion database. Let me know if that works for you!

**Comment by zmek** [comment]
@dvanoni Awesome. That works for me. Thanks.

**Comment by KB22222** [comment]
Thank you for the reply! Does this populate automatically with each reference added or does it need to be added manually? (If automatic, how do you set this up? I have only been able to do it manually) Thank you for the help!

**Comment by dvanoni** [comment]
@KB22222, once you've added the `Collections` property into your Notion database, it should populate automatically whenever you add a new reference into one of your _monitored_ Zotero collections. If you also want to populate the property for existing items that were previously synced into your database, you can use the "Sync to Notion" context menu item that's available in Zotero by right-clicking on either a collection or item(s) in the main Zotero window.

Let me know if that works!

**Comment by KB22222** [comment]
Thank you so much, that was very helpful!

**Comment by meteore** [comment]
Hi @dvanoni -- I think I can make a powerful case for multiple notion databases, as you drafted multiple times in #32 . The key idea is **permissions**, and it has presumably not been mentioned by commenters because they are running _single-user_ Notion instances. But consider the following scenario: you cooperate with different groups of people, e.g. in teamspaces, or you have one _public_ references table that is published to the web and another that is internal to you/your group.

Until and unless Notion supports row-level permissioning*

**Comment by sohaamir** [comment]
I seem to have an issue where my Zotero Collections are synced up to my Notion database and are integrated just fine, however Collections (a multi-select property) doesn't populate within my database. Not sure if I am the only one with this problem?

**Comment by dvanoni** [comment]
Hi @sohaamir, would you be able to share a screenshot of the configuration for your **Collections** property in Notion? I'm wondering if there's something slightly off that's causing Notero to not sync properly.

**Comment by sohaamir** [comment]
Hi @dvanoni, thanks for the quick response! I have attached two screenshots, one is of the general structure of my Notion database, the other is of the Collections property. I feel that the cause is something basic and in plain sight (as it usually is!).

<img width="608" alt="Screenshot 2023-10-18 at 09 11 43" src="https://github.com/dvanoni/notero/assets/35841800/064b696b-3f9f-43df-8f81-649dd1d9b02a">
<img width="1535" alt="Screenshot 2023-10-18 at 09 12 33" src="https://github.com/dvanoni/notero/assets/35841800/27d530f2-79f8-4eef-89f0-8594f214884a">

**Comment by dvanoni** [comment]
@sohaamir, you're right! 😄 It's just a small typo. The property name needs to be plural. It should be `Collections` instead of `Collection`.

I'm hoping to make this more dynamic in the future so that the database properties can be named however you prefer, and you can specify mappings between Zotero fields and Notion properties.

**Comment by sohaamir** [comment]
Duh! Not sure how I didn't see that! Thanks for the help and for developing the tool 😄

**Comment by mdecroly** [comment]
> 
> Would #30 and filtered database views provide an adequate solution for you?

Hi @dvanoni, thank you for this solution which is very useful. However, I still have some issues having two separate database. For context, I would like to have two different database : one for papers related to my thesis and another one for papers related to my intership work. I tried to copy my original database (used for my thesis) that is already linked to zotero in an other page, and filtered the view to have only the collections I wanted. However, even though I copied the original database, it has a different ID, and thus new papers in Zotero are not synced with the copy of the original database. Also, if I use a synced block, I cannot have different filters in each page, since it automatically changes my filter for both databases. Would you have any solutions to have two dabatases which can be both synced with zotero?

**Comment by dvanoni** [comment]
Hi @mdecroly, instead of duplicating the database, it sounds like you want to use a [linked database](https://www.notion.so/help/data-sources-and-linked-databases).

I created a short video below to show how you can use this. I start with my original database, and then I go to a page where I want to add a linked database as a child of that page. I create a new page and start with the "table" option. Then, I select my original database as the source for this new page (i.e. the linked database) and add the filter I want to use for the linked database. The filters on the linked database should stay separate from the filters on your original database.

Let me know if this helps!

https://github.com/dvanoni/notero/assets/299357/1e3cd6f7-9956-4435-9cd0-1f08f8fe7b84

**Comment by mdecroly** [comment]
Thank you very much, this helps a lot @dvanoni !

**Comment by qiutiantxwd** [comment]
> @KB22222, you should be able to achieve this if you add a **multi-select** property named `Collections` to your Notion database. Let me know if that works for you!

This is a really powerful trick

**Comment by Cooper-Zhong** [comment]
Hi @dvanoni, I’m trying to sync different collections in Zotero to separate databases in Notion for easier management. From what I’ve seen in the discussions, it looks like the current setup only lets you sync all collections into one database in Notion, and then you have to filter by the `collections` property to find what you want. I just want to see if my understanding is correct:)

**Comment by dvanoni** [comment]
Hi @Cooper-Zhong, yes that's correct. I think Notion provides a fair amount of flexibility via database views and linked databases, but I know this doesn't meet everyone's needs.

This is one of the highest-voted feature requests, so rest assured it is on the todo list! 😄

**Comment by Cooper-Zhong** [comment]
Thanks for confirming! Glad to hear it's on the todo list—definitely looking forward to that feature.

**Comment by mizanrahman07** [comment]
Hi, @dvanoni, thanks for building Notero. It's truly helpful for me to make workflow and system of my second brain 😊

Like the others, i'm excited to see this feature

And, i wanna request other feature. Is it possible to make the collections property just showing the spesific collection if it's sub collection? 😊

Based on current version, the collections property shows the path of collection. So, if it's sub collection, the master/super collection will be also shown.

It's problem for me, because if i wanna filter based collection, the look will be not good. 

I organize collection systematically, so i have one collection with many sub collections, and the sub collections also have many sub collections

Thanks in advance

**Comment by dvanoni** [comment]
Hi @mizanrahman07, I'm glad Notero has been helpful!

I broke out your additional feature request into #612 so that we can discuss it separately.
